store/iavl/store.go

```diff
@@ -94,18 +138,41 @@ func (st *Store) Commit() types.CommitID {
 		panic(err)
 	}
 
-	return types.CommitID{
+	st.version = version
+	flushed := st.pruning.FlushVersion(version)
```

Comment:
@AdityaSripal do you have a response to this? Or is this even valid?

Reply:
Yes I agree, didn't add this info to `tree.SaveVersion` because I wanted to keep the IAVL API intact.

Doing this would require a breaking change to IAVL in a new major version. Don't want this PR to hold up on that so i think this solution works for now. And we can create an issue to do this on iAVL